Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -37,21 +36,23 @@ def change_rpath(file, old_prefix, new_prefix)
     lib_path = "#{new_prefix}/lib"
     rpath << lib_path unless rpath.include? lib_path
     new_rpath = rpath.join(":")
-    cmd = [patchelf, "--force-rpath", "--set-rpath", new_rpath]
 
-    if file.with_interpreter?
-      old_interpreter = Utils.safe_popen_read(patchelf, "--print-interpreter", file).strip
-      new_interpreter = if File.readable? "#{new_prefix}/lib/ld.so"
+    return unless file.with_interpreter?
+
+    old_interpreter = patchelf.interpreter
+    new_interpreter =
+      if File.readable? "#{new_prefix}/lib/ld.so"
         "#{new_prefix}/lib/ld.so"
       else
         old_interpreter.sub old_prefix, new_prefix
       end
-      cmd << "--set-interpreter" << new_interpreter if old_interpreter != new_interpreter
-    end
 
-    return if old_rpath == new_rpath && old_interpreter == new_interpreter
+    # don't check for (old_interpreter == new_interpreter), patchelf.save handles it
+    return if old_rpath == new_rpath
```

Comment:
I'm a bit concerned that if the RPATH did not change but the interpreter did change, then it would incorrectly return early, when it should in fact patch the ELF file.

Reply:
good catch.